### PR TITLE
Migrate jaeger to otel API

### PR DESCRIPTION
**Pull request by kmjayadeep**
# Changes

OpenTelemetry has deprecated the use of jaeger exporter directly and it has been removed from the library since 1.20 ([changelog](https://github.com/open-telemetry/opentelemetry-go/releases/tag/v1.20.0))

This PR migrates the code to make it compatible with the recommended oltptracehttp exporter instead. Also fixes issues with broken imports when upgrading to otel 1.21

Fixes https://github.com/tektoncd/pipeline/issues/7464

/kind bug

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) if any changes are user facing, including updates to minimum requirements e.g. Kubernetes version bumps
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings). See some examples of [good release notes](https://github.com/tektoncd/community/blob/main/standards.md#good-release-notes).
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/tracing/tracing.go [hunk 1]
```diff
@@ -144,14 +151,34 @@ func (t *tracerProvider) reinitialize() {
 
 func createTracerProvider(service string, cfg *config.Tracing, user, pass string) (trace.TracerProvider, error) {
 	if !cfg.Enabled {
-		return trace.NewNoopTracerProvider(), nil
+		return noop.NewTracerProvider(), nil
+	}
+	u, err := url.Parse(cfg.Endpoint)
```
Comment (kmjayadeep) [inline pkg/tracing/tracing.go:156]: Jaeger exporter expected a complete URL path like `http://website.com:port/path`, while otlp exporter accepts  host and path as separate arguments. This URL parsing is added to ensure that user experience has not changed.

**Comment by afrittoli** [comment]
@kmjayadeep Could you please rebase this PR, so we merge it in time for v0.58?

**Comment by kmjayadeep** [comment]
@afrittoli The PR branch is rebased with main

File: pkg/tracing/tracing.go [hunk 2]
```diff
@@ -150,12 +153,32 @@ func createTracerProvider(service string, cfg *config.Tracing, user, pass string
 	if !cfg.Enabled {
 		return noop.NewTracerProvider(), nil
 	}
+	u, err := url.Parse(cfg.Endpoint)
```
Comment (dibyom) [inline pkg/tracing/tracing.go:156]: @kmjayadeep question: why do we need the additional configuration and the config-tracing configmap when it seems like otel can be configured with env vars (https://pkg.go.dev/go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp#pkg-overview)?

Should we just have one way to configure tracing? (the env vars seem to support more options too e.g. client certs etc. vs the config map which only supports basic auth)

File: pkg/tracing/tracing.go [hunk 3]
```diff
@@ -150,12 +153,32 @@ func createTracerProvider(service string, cfg *config.Tracing, user, pass string
...
+	}
+
+	if user != "" && pass != "" {
+		creds := fmt.Sprintf("%s:%s", user, pass)
+		enc := base64.StdEncoding.EncodeToString([]byte(creds))
+		o := otlptracehttp.WithHeaders(map[string]string{
+			"Authorization": fmt.Sprintf("Basic %s", enc),
+		})
+		opts = append(opts, o)
+	}
```
Comment (afrittoli) [inline pkg/tracing/tracing.go:177]: Is it an option to have a user with empty password?

File: pkg/tracing/tracing.go (hunk 2 above)
Comment (kmjayadeep) [reply pkg/tracing/tracing.go:156]: Hi @dibyom, we have used env vars initially. Then we decided to move to configmap to make it consistent with how other configurations are managed in the project. Also, env variables requires the user to restart the controller when the configuration needs to be changed.
Env variables are still supported. We can also add more parameters in the configmap as needed.

File: pkg/tracing/tracing.go (hunk 3 above)
Comment (kmjayadeep) [reply pkg/tracing/tracing.go:177]: What I found during testing with jaeger and nginx is that if the username or password is configured to be empty, the server just ignores the whole authorization header. So no need to pass the header in code.

**Comment by chitrangpatel** [comment]
Not sure why the test is failing. Seems like a glitch. It is unable to run the second Task I think: https://dashboard.dogfooding.tekton.dev/#/namespaces/tekton-ci/pipelineruns/run-go-coverage-hswz7?pipelineTask=split-full-repo-name&step=split-name

**Comment by kmjayadeep** [comment]
@chitrangpatel I have raised an identical PR in https://github.com/tektoncd/pipeline/pull/7748. Please feel free to close this one if that one gets through

**Pull request merged**